client/my-sites/stats/activity-log/utils.js

```diff
@@ -24,3 +29,56 @@ export function adjustMoment( { timezone, gmtOffset, moment } ) {
 	}
 	return moment;
 }
+
+/**
+ * Accepts an object which contains a string date representation and optionally timezone or offset.
+ * Returns a moment object which is the result of parsing the string adjusted for the timezone
+ * or offset, in that order and if provided.
+ *
+ * @param  {string}  _.startDate Date string representing start of the month (YYYY-MM-DD).
+ * @param  {?string} _.timezone  Timezone representation to apply.
+ * @param  {?string} _.gmtOffset Offset to apply if timezone isn't supplied.
+ * @return {Object}              Start of period moment, adjusted according to timezone or gmtOffset if provided.
+ */
+export function getStartMoment( { gmtOffset, startDate, timezone } ) {
+	if ( timezone ) {
+		if ( ! startDate ) {
+			return momentLib().tz( timezone );
+		}
+
+		return momentLib.tz( startDate, timezone );
+	}
+
+	if ( null !== gmtOffset ) {
+		return momentLib
+			.utc( startDate )
+			.subtract( gmtOffset, 'hours' )
+			.utcOffset( gmtOffset );
+	}
+
+	return momentLib.utc( startDate );
+}
+
+/**
+ * Accepts an object which contains a string date representation and optionally timezone or offset.
+ * Returns an object which is Activity Log query based on the inputs.
+ *
+ * @param  {string}  _.startDate Date string representing start of the month (YYYY-MM-DD).
+ * @param  {?string} _.timezone  Timezone representation to apply.
+ * @param  {?string} _.gmtOffset Offset to apply if timezone isn't supplied.
+ * @return {Object}              Start of period moment, adjusted according to timezone or gmtOffset if provided.
+ */
+export function getActivityLogQuery( { gmtOffset, startDate, timezone } ) {
+	const startMoment = getStartMoment( { gmtOffset, startDate, timezone } );
+	return {
+		dateEnd: startMoment
+			.clone()
+			.endOf( 'month' )
+			.valueOf(),
+		dateStart: startMoment
+			.clone()
+			.startOf( 'month' )
+			.valueOf(),
+		number: 1000,
```

Comment:
Is this going to change when we introduce paging?

Reply:
That's very likely, but I'm not sure what that will look like yet.